ui/src/notifications/endpoints/components/EndpointTypeDropdown.tsx

```diff
@@ -10,11 +10,13 @@ import {extractBlockedEndpoints} from 'src/cloud/utils/limits'
 
 // Types
 import {NotificationEndpointType, AppState} from 'src/types'
+import {isFlagEnabled} from 'src/shared/utils/featureFlag'
 
 interface EndpointType {
   id: NotificationEndpointType
   type: NotificationEndpointType
   name: string
+  flag?: string // feature flag that enable/disable endpoint type
```

Comment:
it feels odd to me to add this to the type. I've made suggestions below for something that feels a little better. let me know if it makes sense!

Reply:
@ebb-tide there are more new notifications types in the game (teams, sensu, discord, opsgenie) , it seems better to me to store the name of the flag in the endpoint descriptor type to avoid branching for every new type. Your proposed code will extend the `if type==='typename` branching in the code with every new type. It does not seem better to me, but I can get myself easily convinced that it is, since I don't know/own the whole concept herein. Can you pls comment on this?

Reply:
OK, I did it your way herein.